packages/atomic/src/components/search/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx

```diff
@@ -156,7 +156,7 @@ export class AtomicSearchBoxRecentQueries {
   private renderItem(value: string): SearchBoxSuggestionElement {
     const query = this.bindings.searchBoxController.state.value;
     return {
-      key: `recent-${cleanUpString(value)}`,
+      key: `recent-${hashString(value)}`,
```

Comment:
Did you encounter some kind of error when the key wasn't cleaned? As far as I'm aware, this isn't an HTML attribute, so I would expect any string to work.

Could we use `value` directly?

Reply:
I had errors when using the instant results `uniqueId`, I can't remember exactly what were the characters that made it break. That's why I thought it safer clean up, but `cleanUpString` also broke when you had recent queries like `['dress', 'dress=']`

Reply:
* The `cleanUpString` function that was used in this file seems to be used in other places in the codebase, for things that will ultimately be output in the DOM as an attribute. So it means it could break in all those other places too ? We should probably fix it everywhere.

* In `cleanUpString`, instead of trying to match on character that can be allowed, would it work to just match on anything not `A-Za-z` and replace with '' ?

```
export function cleanUpString(str: string) {
  return str.replace(/[^a-zA-Z]/g, '').toLocaleLowerCase()
}
```

* This type of function could be easily tested with some basic unit tests (does not need to be full blown Cypress tests). There are already some type of unit tests in the project.

The `hashString` function just seems a bit overkill 😉

Reply:
yeah, I agree that is seems a bit overkill... but your suggestion doesn't really fix it, as it's essentially what it currently does but excluding numbers and spaces.. so you would actually have more clashes than before. for example

```
                                 current func                 your func
['dress', 'dress=']      =>   ['dress', 'dress']          ['dress', 'dress'] 
['dr ess', 'dress=']     =>   ['dr-ess', 'dress']         ['dress', 'dress'] 
['dres s34', 'dre ss']   =>   ['dres-s34', 'dre-ss']      ['dress', 'dress'] 
```

Also, the reason I left instant results and query suggestions the way they were (using `cleanUpString`) is that those values don't come from the user - instant results uses the uniqueId (for which we know what to expect) and query suggestions will hopefully send you valid and 'clean' values that won't clash, so I expect that to be safe. Having said that, bugs lie where you least expect them, so might as well replace it for them all

Reply:
uh, I wouldn't expect `key` to be output in the DOM. First and foremost, I would recommend trying it to see if it works, and if it doesn't, this might potentially be a bug with Stencil?

As a last resort, instead of hashing the string, you could take the raw string and replace spaces with something like `%20`.

Reply:
Wasn't it cleaned up because querySelector was used on it or something like that?

Reply:
ok guys, I think I have a good solution
![Screenshot 2022-06-28 at 15 58 37](https://user-images.githubusercontent.com/30511433/176214286-8d82ef61-4841-4214-aa03-f33eefb8dcf5.png)
as you can see, quite a few characters break the selector
below are a few of the encoding functions and their results:
`cyrb53` works fine
`btoa` is deprecated (could be replaced by `buf.toString('base64')`, but as you can see it sometimes outputs `=`s which don't work
`encodeURIComponent` which is meant for URLs doesn't work because it has `%`s.
`hashIt` (copyright pending) seems to work alright - implementation below

![Screenshot 2022-06-28 at 16 07 53](https://user-images.githubusercontent.com/30511433/176214458-ba5b6746-9a70-490d-94dd-959a2e93113b.png)

```js
function hashIt(str) {
  return str
    .split("")
    .map((ch) => (ch.match(/(\d|\w)+/g) ? ch : ch.charCodeAt(0)))
    .join("");
}
```

what do you guys think? we could call it `encodeForDomAttribute` or something like that, as it's not really hashed

Reply:
Alright! I don't get all the hashing hate then ! 😄

Reply:
Solutions look good!

Reply:
udpated!